### PR TITLE
[SPARK-25385][BUILD] Upgrade Hadoop 3.1 jackson version to 2.7.8

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Upgrade Hadoop 3.1 jackson version to 2.7.8 to fix `JsonMappingException: Incompatible Jackson version: 2.7.8` as Hadoop 3.1 using 2.7.8: https://github.com/apache/hadoop/blob/release-3.1.0-RC1/hadoop-project/pom.xml#L72 .

## How was this patch tested?

manual tests:
```sh
export SPARK_PREPEND_CLASSES=true
build/sbt clean package -Phadoop-3.1
spark-shell
scala> spark.range(10).write.mode("overwrite").parquet("/tmp/spark/parquet")
```

### CODE REVIEW

**Review by felixcheung — APPROVED** [review]
LGTM, changes are only for 3.1 profile

**Comment by felixcheung** [comment]
do we have jenkins tests for 3.1 profile?

**Comment by HyukjinKwon** [comment]
@wangyum, are you doubly sure if still SBT uses the Jackson? I roughly tried this a bit of while ago and found SBT doesn't pick up Maven's.

**Comment by HyukjinKwon** [comment]
Also, IIRC, https://github.com/apache/spark/pull/21596 change is needed for Jackson upgrade.

**Comment by HyukjinKwon** [comment]
Also, I think we should fix https://github.com/apache/spark/pull/21588 first.

File: pom.xml [hunk 1]
```diff
@@ -2694,6 +2694,8 @@
         <hadoop.version>3.1.0</hadoop.version>
         <curator.version>2.12.0</curator.version>
         <zookeeper.version>3.4.9</zookeeper.version>
+        <fasterxml.jackson.version>2.7.8</fasterxml.jackson.version>
+        <fasterxml.jackson.databind.version>2.7.8</fasterxml.jackson.databind.version>
```
Comment (HyukjinKwon) [inline pom.xml]: Please test and check SBT's dependency tree as well.

**Comment by jerryshao** [comment]
Jackson version below 2.9.5 has CVE issues, I would suggest to upgrade to 2.9.6 as #21596 did.

**Comment by wangyum** [comment]
I did a simple test for 2.9.6. It works well. But that pr for 3.0. It means that a simple test on branch 2.4 will fail:
```scala
scala> spark.range(10).write.parquet("/tmp/spark/parquet")
com.fasterxml.jackson.databind.JsonMappingException: Incompatible Jackson version: 2.7.8
  at com.fasterxml.jackson.module.scala.JacksonModule$class.setupModule(JacksonModule.scala:64)
  at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:19)
  at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:730)
  at org.apache.spark.rdd.RDDOperationScope$.<init>(RDDOperationScope.scala:82)
  at org.apache.spark.rdd.RDDOperationScope$.<clinit>(RDDOperationScope.scala)
```
How about merge this pr to branch-2.4 only?

**Comment by jerryshao** [comment]
Do we officially support hadoop3 in branch 2.4? If branch 2.4 doesn't target to support Hadoop3 and this fix is only for Hadoop3, then I don't think it is meaningful to have this fix.

**Comment by jerryshao** [comment]
Btw, I don't think we can run current Spark with Hadoop 3.1 without any change.

**Pull request closed without merging**